### ISSUE TITLE
Aiming from a vehicle in 1st person > Stereo mod disabled on PIMAX 5K+

### ISSUE THREAD

**Issue by DaDrop** [LukeRoss00/gta5-real-mod#164]
When I'm driving and aiming from the vehicle I get stereo rendering off.
It forcing me to close one eye but in addition the view vibrates a lot.

I saw on the Readme.md that it was fixed :
"aiming camera: 
- applied a special-case treatment when the player is aiming from a vehicle in 1st person"

I have not seen any comments about this issue, so maybe is due to my HMD Pimax 5k+

Thanks a lot Luke for this amazing mod !!

**Comment by LukeRoss00** [comment]
You're welcome!

The fix you mentioned only applies to the way the camera direction is computed from the headset angles: it is not a change in camera position or stereo separation.

The camera twitching in cars while looking at certain angles and/or aiming is unfortunately normal. The view angle of the game was originally very restricted while inside vehicles, and unlocking it makes the camera go a little crazy sometimes. Certain car models appear to be more sensitive than others to the issue.

I assume that you mean that stereo rendering looks off as in "abnormal", and not in the sense that the image becomes monoscopic (otherwise you wouldn't need to close one eye). Nobody else reported this issue, and I cannot reproduce it, so I assume it's Pimax-specific. It might also be that it was just a temporary glitch, or that it's limited to some car models?

**Comment by DaDrop** [comment]
Yes you're totally right, the stereo rendering looks off as in "abnormal".

I can describe it as "double vision" sort of effect.
Different from the ghosting effect due to movement and framerating.
It is a fixed offset between the images on each eye.

I tried different cars and I had the same problem whatever the angle of view in the vehicle.

It also happens when I get hit by a car, until the character gets up. 
Or when I get out of a vehicle.

In my first messages I spoke of "stereo rendering off" because 
when I toogle it (NUMPAD 2) the same thing occurs.

And by the way, when I toogle VR off ( NUMPAD -) I then have two flattened images superimposed one inside the other at two different scales. All frozen.
But I toogle VR on, all becomes ok again.

It would seem that this problem is specific to pimax.

Maybe the best option would be to launch the game in oculus mode from Pitool. 
But I already had failures with your previous mod versions, and it was in particular due to the rockstar's launcher. Next thing to try again...

Thanks Luke for your response and for allowing us to enjoy one of the best VR experiences available to date.

**Comment by LukeRoss00** [comment]
Out of curiosity, could you check if the same issue occurs during normal gameplay if you switch off the view matrix fix with the  `O` (letter) hotkey? Because the thing that the moments you mentioned have in common (exiting a vehicle, ragdolling, aiming from a vehicle) is that during those situations I have to temporarily suppress the view matrix fix.

If you can reproduce the issue just by turning off the view matrix fix, which means that you're able to control your character freely as it happens, could you experiment a bit (even though I understand that it would be painful on the eyes!) to determine if the wrong stereo depends on where you look, whether your head is tilted to the side, etc.?

**Comment by DaDrop** [comment]
I don't know what is suppose to do, but when I switch off the view matrix I have the same abnormal render with a fixed offset between the images on each eye.

In addition of this (painful lol) double vision, when I turn my head for look around there is no tracking and the view remains focused on the same area. Just the UI elements (map/fps) are relatives to my head movements in 3D space.

When I am inside a vehicle (with the matrix view still switched on off) and I aiming, the render remain the same (abnormal & offset) but the HMD's tracking is back and the view is synchronized with my head movements.

In conclusion, Aiming from a vehicle with or whithout the view matrix achieve the same result.

I hope it's pretty understanding
I don't know if that helps you to understand better the issue.

don't hesitate if you want to test something else

**Comment by LukeRoss00** [comment]
Okay, that's not normal. Are you sure that all other mod options are at their default values (for instance, have you changed the `RealVR.ini` file in any way)? Do you have any other mods that might be interfering with camera control? When you disable the view matrix fix, the only thing that should change is that the horizon will no longer stay horizontal but instead tilt together with your head; you should definitely not lose all tracking.

**Comment by DaDrop** [comment]
OK interesting if it's not normal.

Yes, in RealVR.ini I changed HeadingControl = 1 and ShowFPS = 1. 
I don't think the issue is due to that but I will try without this changes.

Other mods that might be interfering with camera control : 
The mods that I have apparently do not cause problems for others.

I think openCameraV.asi might be interfering, installed with the OPENVI ASI Manager. 
I don't even use it.

I'll test without it and keep you informed.

**Comment by DaDrop** [comment]
I'm really disappointed because I really thought it was openCameraV.asi that would interfere with the camera.
But no it wasn't, and I tried with the default RealVR.ini and nothing.
Maybe I have to try without other mods... but I doubt it because they are essentially [the compatible ones](https://github.com/LukeRoss00/gta5-real-mod/issues/78).

**Comment by LukeRoss00** [comment]
In [another issue](https://github.com/LukeRoss00/gta5-real-mod/issues/169#issuecomment-622388647), user Aikimaniak just reported exactly the same problems as you have when using a Pimax system, while everything is fine with his Oculus headset.

So, the issues are caused by the Pimax driver/runtime, and unfortunately there is nothing I can do about them at the moment. Your best bet might be asking around the Pimax community if anyone has found a combination of Pitool settings that works, or at least works better.

**Comment by DaDrop** [comment]
Thanks Luke for your support !
For Pimax users, [here is the link](https://community.openmr.ai/t/playing-grand-theft-auto-v-using-the-r-e-a-l-vr-mod-at-120hz-in-the-pimax-5k-is-amazing/24519/61?u=dadrop) of my post on pimax's forum.
